simtools/layout/array_layout.py

```diff
@@ -551,68 +353,102 @@ def _try_set_altitude(self, row, tel, table):
+                self._set_telescope_auxiliary_parameters(tel)
             self._try_set_coordinate(row, tel, table, "ground", "position_x", "position_y")
             self._try_set_coordinate(row, tel, table, "utm", "utm_east", "utm_north")
             self._try_set_coordinate(row, tel, table, "mercator", "latitude", "longitude")
             self._try_set_altitude(row, tel, table)
-
             self._telescope_list.append(tel)
 
-    def initialize_array_layout_from_telescope_file(
-        self, telescope_list_file, telescope_list_metadata_file=None, validate=False
-    ):
+        return table
+
+    def _set_telescope_auxiliary_parameters(self, telescope):
         """
-        Initialize the Layout array from a telescope list file.
+        Set auxiliary CORSIKA parameters for a telescope.
 
         Parameters
         ----------
-        telescope_list_file: str or Path
-            Path to the telescope list file.
-        telescope_list_metadata_file: str or Path
-            Path to the telescope list metadata file.
-        validate: bool
-            Validate the telescope list file.
+        telescope: TelescopePosition
+            Instance of TelescopePosition.
 
-        Returns
-        -------
-        astropy.table.QTable
-            Table with the telescope layout information.
         """
-        table = data_reader.read_table_from_file(
-            file_name=telescope_list_file,
-            validate=validate,
-            metadata_file=telescope_list_metadata_file,
-        )
-        self._initialize_corsika_telescope(table.meta)
-        self._initialize_coordinate_systems(table.meta)
-        self._load_telescope_list(table)
 
-        return table
+        if names.get_class_from_telescope_name(telescope.name) == "telescope":
+            _telescope_model_name = self.db.get_telescope_db_name(telescope.name)
+            self._logger.info(
+                f"Reading auxiliary telescope parameters for {telescope.name}"
+                f" (telescope model {_telescope_model_name})"
+            )
+            if _telescope_model_name not in self._auxiliary_parameters:
+                tel_model = TelescopeModel(
+                    site=self.site,
+                    telescope_model_name=_telescope_model_name,
+                    model_version=self.model_version,
+                    mongo_db_config=self.mongo_db_config,
+                    label=self.label,
+                )
+                self._auxiliary_parameters[_telescope_model_name] = {}
+                for para in ("telescope_axis_height", "telescope_sphere_radius"):
+                    self._auxiliary_parameters[_telescope_model_name][para] = (
+                        tel_model.get_parameter_value_with_unit(para)
+                    )
+
+            for key, value in self._auxiliary_parameters[_telescope_model_name].items():
+                telescope.set_auxiliary_parameter(key, value)
```

Comment:
Yes, and they are read in this part using db_handler in TelescopeModel (although db_handler internally routes it to the db_repo_handler and reads these parameters, until added to the db, from the repository).

This solution reads the telescope model once and saves the parameters for all telescopes. I guess this should be fine, or is there something wrong in this implementation?

Reply:
Not the implementation, I am just wondering about having these parameters as auxiliary ones instead of adding them as part of the "nominal" model like I thought we planned. If you prefer to keep it like this, no problem.